bigquery/google/cloud/bigquery/job.py

```diff
@@ -719,8 +732,12 @@ def cancel(self, client=None):
         if self.location:
             extra_params["location"] = self.location
 
+        # TODO: call thorugh client._call_api() and allow passing in a retry?
```

Comment:
I presume we want to call the API uniformly, i.e. always with an optional retry? A few methods still call it directly, thus the comment.

If confirmed, I will create a separate ticket to address this TODO.

Reply:
Yep. Let's be consistent with retries for sure. Please file a separate ticket.

Reply:
Yes.

I don't recall any reason why `_call_api` wasn't used in the first place.

Reply:
Ticket: #10019.